File: Java/Grading Students.java
// https://www.hackerrank.com/contests/world-codesprint-9/challenges/grading

import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        for(int a0 = 0; a0 < n; a0++){
            int grade = in.nextInt();
            System.out.println((grade > 37 && grade % 5 > 2) ? (grade+2)/5*5:grade);
        }
    }
}

File: Scala/Grading Students.scala
// https://www.hackerrank.com/challenges/grading

import java.io.PrintWriter

object Solution {

    /*
     * Complete the gradingStudents function below.
     */
    def gradingStudents(grades: Array[Int]): Array[Int] = {
        /*
         * Write your code here.
         */
        grades.map(x => if(x > 37) (x + 2)/5*5 else x).zip(grades).map(y => Math.max(y._1, y._2))
    }

    def main(args: Array[String]) {
        val scan = scala.io.StdIn

        val fw = new PrintWriter(sys.env("OUTPUT_PATH"))

        val n = scan.readLine.trim.toInt

        val grades = Array.ofDim[Int](n)

        for (gradesItr <- 0 until n) {
            val gradesItem = scan.readLine.trim.toInt
            grades(gradesItr) = gradesItem}

        val result = gradingStudents(grades)

        fw.println(result.mkString("\n"))

        fw.close()
    }
}
